src/osd/PG.cc

```diff
@@ -3803,6 +3801,25 @@ void PG::scrub(ThreadPool::TPHandle &handle)
     return;
   }
 
+  if (!scrubber.must_scrub) {
+    if ((osd->osd->get_osdmap()->test_flag(CEPH_OSDMAP_NODEEP_SCRUB) && scrubber.deep) ||
+        (osd->osd->get_osdmap()->test_flag(CEPH_OSDMAP_NOSCRUB) && !scrubber.deep)) {
+      dout(10) << "scrub -- noscrub or nodeep scrub" << dendl;
+      scrub_clear_state();
+      scrub_unreserve_replicas();
+      unlock();
+      return;
+    }
+  }
+
+  if (scrubber.stop_scrub && scrubber.must_scrub) {
```

Comment:
Couldn't this read:

if (scrubber.stop_scrub) {
  assert(scrubber.must_scrub);